File: ts/256/169.ts
// 摩尔投票法
function majorityElement(nums: number[]): number {
  let n: number = 0, count: number = 0
  for (const v of nums) {
    if (!count) {
      n = v
      count += 1
      continue
    }
    n === v ? count += 1 : count -= 1
  }
  // if (count === 0) return -1 // 不存在的情况
  return n
};

File: py3/256/169.py
from typing import List

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ans, cnt = None, 0
        for v in nums:
            if cnt == 0:
                ans = v
                cnt += 1
                continue
            cnt += (1 if ans == v else -1)

        return ans